### PR TITLE
Delete operations

### PR DESCRIPTION
**Pull request by tpluscode**
Deleting:

- [x] attribute
- [x] source

What else?

### CODE REVIEW

**Comment by tpluscode** [comment]
The first commit is only moving files around. To review: https://github.com/zazuko/data-cube-curation/pull/147/files/5a388592ffc6d128640ce1e3f3083a01ccaca1f5..08b229861565a9a23593099d429cf6f33b82998e

File: api/hydra/sources.ttl [hunk 1]
```diff
@@ -70,3 +70,12 @@ dataCube:Column
                                                                  hydra:title    "Mapped" ;
                                                                  hydra:description
                                                                                 "Indicates that the column is used in a dimension or measure." ] .
+
+_:DeleteSource
+  a                  hydra:SupportedOperation, hydra-box:View, api:DeleteSource ;
```
Comment (martinmaillard) [inline api/hydra/sources.ttl]: Should it also have `schema:DeleteAction`?

File: api/hydra/attributes.ttl [hunk 2]
```diff
@@ -109,6 +109,15 @@ api:GetAttribute
                        code:arguments ( [ a         code:EcmaScript ;
                                           code:link <file:lib/data-cube/attribute#getHandler> ] ) ] .
 
+_:DeleteAttribute
+  a                  hydra:SupportedOperation, hydra-box:View, api:DeleteAttribute ;
```
Comment (martinmaillard) [inline api/hydra/attributes.ttl]: Should it also have `schema:DeleteAction`?

File: api/hydra/sources.ttl (hunk 1 above)
Comment (tpluscode) [reply api/hydra/sources.ttl]: Sure, I don't see why not

**Comment by martinmaillard** [comment]
What happens if you delete a source that is referenced by a table? Will it also delete the table?

**Comment by tpluscode** [comment]
Aha, good thinking.

So, there are two possibilities:

1. A table is mapped from a source and not used otherwise.
2. A source is referenced by other tables

In the order, I would allow the removal and yes, the tables should of course disappear. Will add a test for that.

The latter should probably prevent the operation, right?

**Comment by martinmaillard** [comment]
I'm really not sure. I kinda feel like both cases should be handled in the same way. Either we prevent deleting a source that is used anywhere, or we cascade delete anything related (tables + attributes).

We should probably go for the easiest solution, since it's not really an important use case.

**Comment by tpluscode** [comment]
Then let's simply prevent

File: api/lib/read-graphs/source/isReferenced.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    .where(`
+        ?s ?uses <${sourceId}>
+
+        MINUS {
+            ?s a dataCube:Project
+        }
+        MINUS {
+            <${sourceId}> api:columns ?s
+        }
+    `)
```
Comment (tpluscode) [inline api/lib/read-graphs/source/isReferenced.ts]: Here's a SPARQL to check that the source is referenced.

It excludes reference by the parent project and its columns collection resource. Any other occurrence will prevent the delete

**Pull request merged**